presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcSelectivePageSource.java

```diff
@@ -92,11 +103,35 @@ public boolean isFinished()
     @Override
     public Page getNextPage()
     {
+        boolean supplyRowIDs = recordReader.isColumnPresent(ROW_ID_COLUMN_INDEX);
```

Comment:
Let's move  creation of this flag to constructor, ideally as an argument supplied by the orc page factory